cjworkbench/tests/test_catalogs.py

```diff
@@ -0,0 +1,848 @@
+from django.test import SimpleTestCase
+from babel.messages.catalog import Catalog, Message
+from cjworkbench.i18n.catalogs.util import (
+    message_unique_identifier,
+    find_corresponding_message,
+    find_corresponding_string,
+    find_fuzzy_messages,
+    fill_catalog,
+    mark_fuzzy,
+    remove_strings,
+)
+from cjworkbench.i18n.catalogs.merge import _merge_source_catalog, _merge_catalog
+
+
+class CatalogTest(SimpleTestCase):
+    def assertCatalogsDeeplyEqual(self, catalog: Catalog, other_catalog: Catalog):
+        """ Assert that the two catalogs contain the same messages, where message equality is deep.
+        
+        Ignores header message.
+        """
+        self.assertCatalogDeeplyContainsCatalog(catalog, other_catalog)
+        self.assertCatalogDeeplyContainsCatalog(other_catalog, catalog)
+
+    def assertCatalogDeeplyContainsCatalog(
+        self, catalog: Catalog, other_catalog: Catalog
+    ):
+        """ Assert that `other_catalog` contains all the messages of `catalog`, where message equality is deep.
+        
+        Ignores header message.
+        """
+        self.assertEqual(catalog.locale, other_catalog.locale)
+        for message in catalog:
+            if message.id:  # ignore header
+                other_message = find_corresponding_message(other_catalog, message)
+                self.assertTrue(
+                    other_message, msg=f"Message {message} not found in catalog"
+                )
+                self.assertMessageDeeplyEqual(message, other_message)
+
+    def assertMessageDeeplyEqual(self, message: Message, other_message: Message):
+        self.assertEqual(message, other_message)  # this compares id and context
+        self.assertEqual(message.string, other_message.string)
+        self.assertEqual(message.flags, other_message.flags)
+        self.assertEqual(message.auto_comments, other_message.auto_comments)
+        self.assertEqual(message.user_comments, other_message.user_comments)
+        self.assertEqual(message.locations, other_message.locations)
+
+
+class UtilTest(CatalogTest):
```

Comment:
Why is this a `SimpleTestCase` instead of a plain `unittest.TestCase`?

Reply:
No idea. I just copied from some other test file.

Reply:
Please extend `unittest.TestCase`. It has fewer features. It doesn't depend on Django. (In case it isn't comically clear: I _hate_ depending on Django ;).)